### ISSUE TITLE
Merlin throw syntax error on reason files

### ISSUE THREAD

**Issue by laibulle** [reasonml-editor/vscode-reasonml#117]
Merlin seems to not detect the reason syntax in vscode. But this file build and run without error.

<img width="771" alt="screen shot 2017-10-18 at 9 17 13 am" src="https://user-images.githubusercontent.com/2327626/31705311-36b3fb24-b3e5-11e7-84f9-8c5559a8f161.png">

**Comment by thomhos** [comment]
Same here. Clean install of bs-platform and then a bsb -init foo -theme basic-reason. It compiles, but vscode doesn't recognize the syntax.

**Comment by freebroccolo** [comment]
How do you have your development environment setup? What is the output of `which ocamlmerlin` from the shell? It may be that you somehow have an outdated version of merlin in your PATH when the extension is loaded.

**Comment by thomhos** [comment]
A bit more detail;

1. install reason
```npm install -g https://github.com/reasonml/reason-cli/archive/3.0.0-bin-darwin.tar.gz```
2. install bs-platform
```npm install -g bs-platform```
3. install reason for vscode
4. setup a new project
```bsb -init my-proj -theme basic-reason```
5. use terminal to start vscode
```code .```

When i run ```which ocamlmerlin refmt ocamlmerlin reason``` i get:
```
/Users/thomhos/.nvm/versions/node/v8.8.1/bin/ocamlmerlin
/Users/thomhos/.nvm/versions/node/v8.8.1/bin/refmt
/Users/thomhos/.nvm/versions/node/v8.8.1/bin/ocamlmerlin
```

Note, I am using nvm to manage node versions. Not sure if that is related to the issue.

In VSCode I have 2 issues;

1. An error from merlin saying the first character is not allowed (there doesn't seem to be any character that triggers this.
<img width="1136" alt="screen shot 2017-10-30 at 19 55 43" src="https://user-images.githubusercontent.com/11661846/32189995-a6f44d46-bdac-11e7-8f01-eace51e776fa.png">

2. When I add ```"refmt": 3``` to the bsconfig it won't build anymore.
```
> bsb -make-world

ninja: Entering directory `lib/bs'
[2/3] Building src/demo.mlast
FAILED: src/demo.mlast
/Users/thomhos/.nvm/versions/node/v8.8.1/lib/node_modules/bs-platform/bin/bsc.exe -pp "/Users/thomhos/.nvm/versions/node/v8.8.1/lib/node_modules/bs-platform/bin/refmt3.exe --print binary"    -w -30-40+6+7+27+32..39+44+45+101 -nostdlib -I '/Users/thomhos/Code/personal/my-proj/node_modules/bs-platform/lib/ocaml' -no-alias-deps -color always -c -o src/demo.mlast -bs-syntax-only -bs-binary-ast -impl /Users/thomhos/Code/personal/my-proj/src/demo.re
File "/Users/thomhos/Code/personal/my-proj/src/demo.re", line 3, characters 7-40:
Error: 2114: syntax error, consider adding a `;' before
File "/Users/thomhos/Code/personal/my-proj/src/demo.re", line 1:
Error: Error while running external preprocessor
Command line: /Users/thomhos/.nvm/versions/node/v8.8.1/lib/node_modules/bs-platform/bin/refmt3.exe --print binary '/Users/thomhos/Code/personal/my-proj/src/demo.re' > /var/folders/t3/19hsyn792cx8b59kl28jw3c40000gn/T/ocamlpp85a406

ninja: error: rebuilding 'build.ninja': subcommand failed
```

I'm finding it a bit hard to locate the origin of the issue, if it's bs-platform, reason-cli or anything else?

Also, it seems to be calling some .exe files, which i'm not sure about considering I'm on mac.

Thanks!

**Comment by freebroccolo** [comment]
@laibulle @thomhos Is this still happening with the latest release? Do you have OCaml installed separately outside of `bs-platform`?

**Comment by thomhos** [comment]
@freebroccolo I'm trying a re-install. But it seems that installing bs-platform from source is not working at the moment on npm@5. I'll try again when that seems to be resolved.

**Comment by laibulle** [comment]
@thomhos Actually everything works since I installed reason via opam.

**Comment by thomhos** [comment]
I'm getting a bunch of errors after doing that. But I'm attending a meetup in Amsterdam next week about reason (inc workshop) so I'm hoping someone can help me get set up there :)

**Comment by freebroccolo** [comment]
@thomhos If you installed new stuff through opam, make sure you completely quit the editor and restart it. If you were launching it from the terminal, you will probably need to restart that too so that the environment is properly updated.

**Comment by jessep** [comment]
FYI, for others having these issues, I was having this same issue, and they're fixed now. Here's what I did.

1) I just uninstalled and re-installed bs-platform and reason-cli. I realized that I had installed ocaml through homebrew (I'm on a mac) and uninstalled it. Every time I made a change, I would do `code .` in the project directory, and the same problem would persist.
2) I reread this thread, and realized I needed to open a new tab in my terminal before typing `code .`.
3) It worked. 

Huzzah. This problem was giving me that sinking feeling.
